lineplot/lineplot.app.js

```diff
@@ -0,0 +1,231 @@
+                                            });
+                                        }
+                                    } else {
+                                        // else add warning alert
+                                        AlertsService.addAlert(err.data, 'warning');
+                                    }
+                                }
+                            });
+                        });
+                    }
+                }
+            }])
+            .controller('LineplotController', ['AlertsService', 'dataFormats', 'LineplotUtils', 'UriUtils', '$http', '$window', '$rootScope', '$scope', '$timeout', function LineplotController(AlertsService, dataFormats, LineplotUtils, UriUtils, $http, $window, $rootScope, $scope, $timeout) {
+                var vm = this;
+                vm.alerts = AlertsService.alerts;
+                vm.dataFormats = dataFormats;
+                vm.x_label = lineplotConfig.x_axis_label;
+                vm.model = {
+                    user: $rootScope.user,
+                    subject: $rootScope.subject_id,
+                    duration: $rootScope.duration,
+                    date: moment($rootScope.start_time).format(dataFormats.date),
+                    time: moment($rootScope.start_time).format(dataFormats.time24),
+                    meridiem: 'AM',
+                    type: 'Line Plot',
+                };
+
+                vm.applyDatetime = function () {
+                    var timestamp = moment(vm.model.date + vm.model.time, dataFormats.date + dataFormats.time12).format(dataFormats.datetime.submission);
+                    if (vm.model.duration != null && vm.model.duration.length > 0) {
+                    	$rootScope.duration = vm.model.duration;
+                    }
+                    LineplotUtils.getData(timestamp);
+                }
+
+                vm.toggleMeridiem = function () {
+                    var meridiem = vm.model.meridiem;
+                    if (meridiem.charAt(0).toLowerCase() === 'a') {
+                        return meridiem = 'PM';
+                    }
+                    return meridiem = 'AM';
+                }
+
+                vm.applyCurrentDatetime = function () {
+                    vm.model.date = moment().format(dataFormats.date);
+                    vm.model.time = moment().format(dataFormats.time24);
+                }
+
+                vm.removeValue = function () {
+                    vm.model.date = null;
+                    vm.model.time = null;
+                }
+
+
+                vm.plotsLoaded = false;
+                $scope.showLineplot = function () {
+                	if (vm.model.user == null) {
+                		vm.model.user = $rootScope.user;
+                	}
+                    $scope.$apply(function () {
+                        vm.plotsLoaded = true;
+                    });
+                }
+
+                vm.logout = function () {
```

Comment:
Why does this function need to be re-written? Was there an issue with using `Session.logout()`?